code/script.js

```diff
@@ -229,97 +389,139 @@ const setSecret = () => {
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  } else if (category === 'hair') {
+    if (keep) {
+      alert(`Yes, the champion has ${value} hair! Keep all champions with ${value} hair`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else {
+      alert(`No, the champion doesn't have ${value} hair! Remove all champions that have ${value} hair`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      }
+     } else if(category === 'costume') {
+      if (keep) {
+        alert(`Yes, the champion has ${value} costume! Keep all champions with ${value} costume`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+      } else {
+        alert(`No, the champion doesn't have ${value} costume! Remove all champions that have ${value} costume`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      }
+    }
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+    
+    generateBoard()
+  }
 
-  // Invoke a function to redraw the board with the remaining people.
-}
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+// Player can confirm their guess when clicking the guess button, it then stores the value and invokes function to check their guess.
+const guess = (suspectCharacter) => {
+  
+  const confirmGuess = confirm(`Is this champion your guess?`)
+  if (confirmGuess)
+  checkMyGuess(suspectCharacter)
+  
 }
 
-// If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+// If you confirm your guess, this function is invoked to display either if it was correct or false.
+const checkMyGuess = (suspectCharacter) => {
+
+  if (suspectCharacter === secret.name) {
+  winOrLoseText.innerHTML = `Victory! You guessed on ${suspectCharacter} which is the right champion!`
+  victorySound.play()
+} else {
+  winOrLoseText.innerHTML = `Defeat! The right champion was ${secret.name}...`
+  defeatSound.play()
+}
+  
+  winOrLose.style.display = 'flex';
+  
```

Comment:
nice! the winOrLose div is shown but did you hide the game board?
Bigger question is do we need to hide it if we cant see it anyway? 
Referring to line 318.

Reply:
Not sure what you are referring to in line/row 318, it's inte the middle of an object :D do you mean the javascript or is it another row your thinking about?

The code on row 505 displays the win or loose wrapper. If I remove it and try to guess on a character it won't appear..

Reply:
I think I was unclear. But I'm referring to line 505/506 in your code and the row 318 is from the starter code I think.
In the "starter code" they had a comment that said something like "don't forget to show the winorlose screen and also hide the gameboard".

So I was thinking that maybe there should be a `board.style.display ="none"` together with the `winOrLose.style.display = 'flex'`.

So thats what I meant with my second statement about do we really need it because it seems good to just make the winorlose screen appear.